### PR TITLE
Handle list data type

### PR DESCRIPTION
**Pull request by ollietulloch**
This PR is an initial commit of handling the more complex `:list` arrow data type.

If the `arrow_data_type` defined in the mapping is `:list`, it is expected that an  `arrow_list_field`  is also defined, which will include `name`, `type`, and `split`.

`name` is the label given to each item in the list
`type` is the arrow field type for each item in the list, e.g. `:int32`, `:string`.
`split` is what is used to split the incoming string into an array

Currently, the `type` needs to be the same for all items in a list.  This doesn't yet support nested lists.

Errors could be handled more gracefully where there is missing config in the mapping

### CODE REVIEW

**Comment by timgentry** [comment]
Looks good to me. @joshpencheon?

**Comment by timgentry** [comment]
My only question was whether or not you wanted to handle commas in values, we discussed `decode: csv`, but pragmatically I'd be guided by your current use case.

File: lib/ndr_parquet_generator.rb [hunk 1]
```diff
@@ -99,7 +99,11 @@ def arrow_field_types(table)
           column['mappings'].each do |mapping|
             field = mapping['field']
             arrow_data_type = mapping['arrow_data_type'] || :string
-            field_types[klass][field] = arrow_data_type
+            if arrow_data_type == :list
+              field_types[klass][field] = mapping['arrow_list_field'].symbolize_keys
```
Comment (joshpencheon) [inline lib/ndr_parquet_generator.rb]: Might you want to “fetch” the value here, so if it isn’t provided you get a helpful error?

File: test/resources/national_collection.yml [hunk 2]
```diff
@@ -28,7 +28,11 @@
   - column: SQU03_6_2_2:N
     mappings:
     - field: SQU03_6_2_2
-      arrow_data_type: :int32
+      arrow_data_type: :list
```
Comment (joshpencheon) [inline test/resources/national_collection.yml:31]: I still do wonder if a format identifier like `list[int]` or `[:int]` might not be a bit more idiomatic here, although granted would probably cause other internal issues with the symbol representation. The latter would still be valid YAML I think, representing a literal array.

**Comment by ollietulloch** [comment]
@timgentry the use case I've come across is a `;` separated string e.g. "code_a;code_b;code_c". This would have a mapping something like:

 ```
      arrow_data_type: :list
      arrow_list_field:
        name: code_name
        type: :string
        split: ';'

```

File: lib/ndr_parquet_generator.rb (hunk 1 above)
Comment (ollietulloch) [reply lib/ndr_parquet_generator.rb]: Thanks @joshpencheon - 3b4c769

File: test/resources/national_collection.yml (hunk 2 above)
Comment (ollietulloch) [reply test/resources/national_collection.yml:31]: @joshpencheon would you have something like:

```
arrow_data_type: [:string]
arrow_list_field:
  name: code_name
  split: ';'
```

**Comment by timgentry** [comment]
Sorry @ollietulloch I had to get some changes in that were blocking the lambda work, so you'll need to merge those changes into your branch.

**Comment by ollietulloch** [comment]
@timgentry - no probs, should be done in 28d1ecc

File: test/resources/national_collection.yml (hunk 2 above)
Comment (joshpencheon) [reply test/resources/national_collection.yml:31]: That's a good point; I think I'd been thinking along the lines of this being an expression of datatype, whereas actually it's more like a processing directive so perhaps discount my earlier feedback!

**Pull request merged**